src/pages/patientView/clinicalInformation/ClinicalInformationMutationalSignatureTable.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from "react";
+import {ClinicalAttribute, ClinicalData} from "../../../shared/api/generated/CBioPortalAPI";
+import LazyMobXTable, {Column} from "shared/components/lazyMobXTable/LazyMobXTable";
+import styles from './style/patientTable.module.scss';
+import {SHOW_ALL_PAGE_SIZE} from "../../../shared/components/paginationControls/PaginationControls";
+import {IMutationalSignature} from "../../../shared/model/MutationalSignature";
+import convertSamplesData, {IConvertedSamplesData} from "./lib/convertSamplesData";
+import _ from 'lodash';
+import {labelMobxPromises} from "mobxpromise";
+
+export interface IClinicalInformationMutationalSignatureTableProps {
+    data: IMutationalSignature[];
+    showTitleBar: boolean;
+    cssClass?:string;
+    showFilter?:boolean;
+    showCopyDownload?:boolean;
+}
+
+class MutationalSignatureTable extends LazyMobXTable<IMutationalSignatureRow> {}
+
+interface IMutationalSignatureRow {
+    mutationalSignatureId:string;
+    sampleValues:{[uniqueSampleKey: string]: number},
+
+};
+
+interface IColumns{
+    id:string
+}
+
+export function prepareDataForTable(mutationalSignatureData: IMutationalSignature[]):IMutationalSignatureRow[] {
+
+    const tableData: IMutationalSignatureRow[] = [];
+
+    //group data by mutational signature
+    //[{id: mutationalsignatureid, samples: [{}, {}]}]
+    let sampleInvertedDataByMutationalSignature:Array<any> = _(mutationalSignatureData)
+        .groupBy(mutationalSignatureSample => mutationalSignatureSample.mutationalSignatureId)
+        .map((mutationalSignatureSampleData, mutationalSignatureId)  => ({id: mutationalSignatureId, samples: mutationalSignatureSampleData}))
+        .value();
+
+    for (const mutationalSignature of sampleInvertedDataByMutationalSignature){
+        let mutationalSignatureRowForTable: IMutationalSignatureRow = {mutationalSignatureId:"", sampleValues: {}};
+        mutationalSignatureRowForTable.mutationalSignatureId = mutationalSignature.id;
+        for (const sample of mutationalSignature.samples){
+            mutationalSignatureRowForTable.sampleValues[sample.sampleId] = Number(sample.value.toFixed(4));
+        }
+        tableData.push(mutationalSignatureRowForTable);
+    }
+    return tableData;
+}
+
+export function prepareDataForColumns(mutationalSignatureData: IMutationalSignature[]): IColumns[]{
+    return _.map(_.uniqBy(mutationalSignatureData,"sampleId"), (uniqSample) => ({id: uniqSample.sampleId}));
```

Comment:
This now needs to be grouped by unique sample id. And also other places using sample id as the key.